quantum/nodes/bloch-sphere/bloch-sphere.js

```diff
@@ -0,0 +1,109 @@
+'use strict';
+
+const snippets = require('../../snippets');
+const shell = require('../../python').PythonShell;
+const errors = require('../../errors');
+
+module.exports = function(RED) {
+  function BlochSphereNode(config) {
+    RED.nodes.createNode(this, config);
+    this.name = config.name;
+    this.qubits = [];
+    this.qreg = '';
+    const node = this;
+
+    this.on('input', async function(msg, send, done) {
+      let qubitsArrived = true;
+
+      let error = errors.validateQubitInput(msg);
+      if (error) {
+        done(error);
+        return;
+      }
+      // Throw Error if:
+      // - The user connects it to a node that is not from the quantum library
+      if (typeof(msg.payload.register) === 'undefined') {
+        node.qreg = undefined;
+        node.qubits.push(msg);
+
+        // Check if all qubits arrived.
+        if (node.qubits.length < msg.payload.structure.qubits) {
+          qubitsArrived = false;
+        }
+      } else {
+        // If the quantum circuit has registers
+        // Keep track of qubits that have arrived and the remaining ones
+        if (node.qubits.length == 0) node.qreg = {};
+
+        // Throw an error if too many qubits are received by the simulator node
+        // because the user connected qubits from different quantum circuits
+        if ((
+          !Object.keys(node.qreg).includes(msg.payload.registerVar) &&
+          Object.keys(node.qreg).length == msg.payload.structure.qreg
+        ) || (
+          Object.keys(node.qreg).includes(msg.payload.registerVar) &&
+          node.qreg[msg.payload.registerVar].count == node.qreg[msg.payload.registerVar].total
+        )) {
+          done(new Error(errors.QUBITS_FROM_DIFFERENT_CIRCUITS));
+          reset();
+          return;
+        }
+
+        // Storing information about which qubits were received
+        if (Object.keys(node.qreg).includes(msg.payload.registerVar)) {
+          node.qreg[msg.payload.registerVar].count += 1;
+        } else {
+          node.qreg[msg.payload.registerVar] = {
+            total: msg.payload.totalQubits,
+            count: 1,
+          };
+        }
+
+        node.qubits.push(msg);
+
+        // Checking whether all qubits have arrived or not
+        if (Object.keys(node.qreg).length == msg.payload.structure.qreg) {
+          Object.keys(node.qreg).map((key) => {
+            if (node.qreg[key].count < node.qreg[key].total) {
+              qubitsArrived = false;
+            }
+          });
+        } else {
+          qubitsArrived = false;
+        }
+      }
```

Comment:
The whole code & logic for the "qubits waiting phase" of "quantum output nodes" has been improved and simplified.

Can you please have a look at "node output template" and try to reuse the same code/structure ?

It would be best for consistency